cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -331,13 +338,16 @@ private void maybeAddLeaderChangeTasks(Collection<ExecutionProposal> proposals,
      * chances to make progress.
      */
     boolean newTaskAdded = true;
+    _interPartMoveBrokerId.addAll(_interPartMoveTaskByBrokerId.keySet());
```

Comment:
Note: `readyBrokers` is replaced by `_interPartMoveTaskByBrokerId` to reduce iteration speed and avoid NPE in `brokerComparator`. 
`readyBroker` which is not in `_interPartMoveTaskByBrokerId` would never be involved in any execution round.